agent/src/capability/servicemanifestfileloader.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2016-2017 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+
+#pragma once
+
+#include <jansson.h>
+
+#include "softwarecontainer-common.h"
+#include "servicemanifestloader.h"
+#include "configstoreerror.h"
+
+namespace softwarecontainer {
+
+class ServiceManifestFileLoader : public ServiceManifestLoader
```

Comment:
The biggest reason for the change was for testability. The other abstract class is inside the unit-test suite (also included in this PR) configstore_unittest.

I do agree that I could add more class documentation though.